File: java/SelectionSort.java
public class SelectionSort {

    public static void selectionSort(int inputArray[]) {

        // local variables for reuse during algorithm
        int lowestValueIndex;
        int swapValue;

        // loop through each element to the right of currentIndex
        for (int currentIndex = 0; currentIndex < inputArray.length; currentIndex++) {
            // hold index of smallest element, initially assign the first index 0
            lowestValueIndex = currentIndex;
            // loop through each element to the right of currentIndex
            for (int nextIndex = currentIndex + 1; nextIndex < inputArray.length; nextIndex++) {
                // if the value of the element to the right (nextIndex) is less than the
                // value of the element at the lowestValueIndex, reassign the  value
                // of lowestValueIndex to the nextIndex
                if (inputArray[nextIndex] < inputArray[lowestValueIndex]) {
                    lowestValueIndex = nextIndex;
                }
            }
            // hold the value at the currentIndex
            swapValue = inputArray[currentIndex];
            // swap the currentIndex and smallestElementIndex values
            inputArray[currentIndex] = inputArray[lowestValueIndex];
            inputArray[lowestValueIndex] = swapValue;
        }
    }

    public static void printArray(int inputArray[]) {
        for (int value : inputArray) {
            System.out.print(value + " ");
        }
    }

    public static void main(String[] args) {

        // define an array to test
        int exampleArray[] = {34, 88, 98, 4, 23, 37, 7, 56, 9, 20};

        System.out.println("Unsorted array:");
        printArray(exampleArray);

        selectionSort(exampleArray); // sort the array

        System.out.println("\nSorted array:");
        printArray(exampleArray);
    }
}

File: csharp/SelectionSort.cs
using System;

public class SelectionSort
{

    public static void SelectSort(int[] inputArray)
    {
        // local variables for reuse during the algorithm
        int swapValue;
        int smallestValueIndex;

        // loop through each element
        for (int currentIndex = 0; currentIndex < inputArray.Length; currentIndex++)
        {
            // hold index of smallest element, initially assign the first index 0
            smallestValueIndex = currentIndex;

            // loop through each element to the right of currentIndex
            for (int nextIndex = currentIndex + 1; nextIndex < inputArray.Length; nextIndex++)
            {
                // if the value of the element to the right (nextIndex) is less than the 
                // value of the element at the smallestElementIndex, reassign the  value 
                // of smallestElementIndex to the nextIndex
                if (inputArray[nextIndex] < inputArray[smallestValueIndex])
                {
                    smallestValueIndex = nextIndex;
                }
            }
            // hold the value at the currentIndex
            swapValue = inputArray[currentIndex];
            // swap the currentIndex and smallestElementIndex values
            inputArray[currentIndex] = inputArray[smallestValueIndex];
            inputArray[smallestValueIndex] = swapValue;
        }
    }

    public static void PrintArray(int[] inputArray)
    {
        foreach (int value in inputArray) {
            Console.Write($"{value} ");
        }
    }
    
    public static void Main()
    {
        // create array for test
        int[] exampleArray = { 34, 88, 98, 4, 23, 37, 7, 56, 9, 20 };

        Console.WriteLine("Unsorted array:");
        PrintArray(exampleArray);

        SelectSort(exampleArray); // sort the array

        Console.WriteLine("\nSorted array:");
        PrintArray(exampleArray);
    }
}